internal/app/backend/backend_service_test.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backend
+
+import (
+	"io"
+	"testing"
+
+	"open-match.dev/open-match/internal/app/frontend"
+
+	"github.com/rs/xid"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"open-match.dev/open-match/internal/app/mmlogic"
+	"open-match.dev/open-match/internal/pb"
+	"open-match.dev/open-match/internal/rpc"
+
+	rpcTesting "open-match.dev/open-match/internal/rpc/testing"
+	statestoreTesting "open-match.dev/open-match/internal/statestore/testing"
+)
+
+func TestAssignTicketsEmptyRequest(t *testing.T) {
+	assert := assert.New(t)
+	tc := createBackendForTest(t)
+	defer tc.Close()
+
+	assignTicketsLoop(tc, &pb.AssignTicketsRequest{}, func(_ *pb.AssignTicketsResponse, err error) {
+		assert.Equal(codes.InvalidArgument, status.Convert(err).Code())
+	})
+	assignTicketsLoop(tc, &pb.AssignTicketsRequest{Assignment: &pb.Assignment{}}, func(_ *pb.AssignTicketsResponse, err error) {
+		assert.Equal(codes.InvalidArgument, status.Convert(err).Code())
+	})
+	assignTicketsLoop(tc, &pb.AssignTicketsRequest{TicketId: []string{xid.New().String(), xid.New().String()}}, func(_ *pb.AssignTicketsResponse, err error) {
+		assert.Equal(codes.InvalidArgument, status.Convert(err).Code())
+	})
+
+	req := &pb.AssignTicketsRequest{
+		TicketId: []string{xid.New().String(), xid.New().String()},
+		Assignment: &pb.Assignment{
+			Connection: "localhost",
+		},
+	}
+	assignTicketsLoop(tc, req, func(resp *pb.AssignTicketsResponse, err error) {
+		assert.Equal(codes.NotFound, status.Convert(err).Code())
+	})
+}
+
+func TestAssignTicketsNormalRequest(t *testing.T) {
+	assert := assert.New(t)
+	tc := createBackendForTest(t)
+	defer tc.Close()
+
+	fe := pb.NewFrontendClient(tc.MustGRPC())
+
+	ctResp, err := fe.CreateTicket(tc.Context(), &pb.CreateTicketRequest{Ticket: &pb.Ticket{}})
+	assert.Nil(err)
+
+	req := &pb.AssignTicketsRequest{
+		TicketId: []string{ctResp.Ticket.Id},
+		Assignment: &pb.Assignment{
+			Connection: "localhost",
+		},
+	}
+	assignTicketsLoop(tc, req, func(resp *pb.AssignTicketsResponse, err error) {
+		assert.Equal(&pb.AssignTicketsResponse{}, resp)
+	})
+
+	gtResp, err := fe.GetTicket(tc.Context(), &pb.GetTicketRequest{TicketId: ctResp.Ticket.Id})
+	assert.Nil(err)
+	assert.Equal("localhost", gtResp.Assignment.Connection)
+}
+
```

Comment:
Same as above - TestFetchMatches as a test - and all test cases as table based tests within it.